packages/gatsby/cache-dir/production-app.js

```diff
@@ -64,6 +64,12 @@ apiRunnerAsync(`onClientEntry`).then(() => {
     if (redirect) {
       pathname = redirect.toPath
     }
+
+    // If we had a service worker update, no matter the path, reload window
+    if (window.GATSBY_SW_UPDATED) {
```

Comment:
@pieh can you check this out when you get a chance? is this essentially what you had in mind?